### PR TITLE
[MODULAR] SmartDart Pistols now show the inserted SmartDarts inside of them.

### PR DESCRIPTION
**Pull request by softcerv**
## About The Pull Request
When examining a SmartDart pistol, it will now show what SmartDart(s) are loaded inside of the chamber.
image

## How This Contributes To The Skyrat Roleplay Experience
This helps contribute to the idea of the SmartDart gun being a Smart version of the syringe gun, while providing useful QoL for medical players. 

## Changelog

:cl:
qol: SmartDart Pistols now show the inserted SmartDarts inside of them when examined.
/:cl:

### CODE REVIEW

**Review by Zonespace27 — CHANGES_REQUESTED** [review]
Do the same for the Honeybee

**Comment by softcerv** [comment]
> Do the same for the Honeybee

the deed has been done.

File: modular_skyrat/modules/gun_cargo/code/objects/guns/interdyne.dm [hunk 1]
```diff
@@ -31,6 +31,9 @@
 	. += "<br>You can fire the underbarrel device with [span_bold("right click")]."
 	. += "<br>You can unload the underbarrel device by [span_bold("right clicking with an empty hand")]."
 
+	for(var/obj/item/reagent_containers/syringe/dart in underbarrel.syringes)
+		. += "There is a [dart] loaded"
```
Comment (Zonespace27) [inline modular_skyrat/modules/gun_cargo/code/objects/guns/interdyne.dm]:
```suggestion
		. += "There is a [dart] loaded."
```

File: modular_skyrat/modules/medical/code/smartdarts.dm [hunk 2]
```diff
@@ -60,6 +60,12 @@
 		to_chat(user, span_notice("The [container] is unable to fit inside of the [src]! Try using a <b>SmartDart</b> instead."))
 		return FALSE
 
+/obj/item/gun/syringe/smartdart/examine(mob/user)
+	. = ..()
+
+	for(var/obj/item/reagent_containers/syringe/dart in syringes)
+		. += "There is a [dart] loaded"
```
Comment (Zonespace27) [inline modular_skyrat/modules/medical/code/smartdarts.dm]:
```suggestion
		. += "There is a [dart] loaded."
```

File: modular_skyrat/modules/medical/code/smartdarts.dm [hunk 3]
```diff
@@ -60,6 +60,12 @@
 		to_chat(user, span_notice("The [container] is unable to fit inside of the [src]! Try using a <b>SmartDart</b> instead."))
 		return FALSE
 
+/obj/item/gun/syringe/smartdart/examine(mob/user)
+	. = ..()
+
+	for(var/obj/item/reagent_containers/syringe/dart in syringes)
```
Comment (tf-4) [inline modular_skyrat/modules/medical/code/smartdarts.dm]:
```suggestion
	for(var/obj/item/reagent_containers/syringe/dart as anything in syringes)
```

File: modular_skyrat/modules/gun_cargo/code/objects/guns/interdyne.dm [hunk 4]
```diff
@@ -31,6 +31,9 @@
 	. += "<br>You can fire the underbarrel device with [span_bold("right click")]."
 	. += "<br>You can unload the underbarrel device by [span_bold("right clicking with an empty hand")]."
 
+	for(var/obj/item/reagent_containers/syringe/dart in underbarrel.syringes)
```
Comment (tf-4) [inline modular_skyrat/modules/gun_cargo/code/objects/guns/interdyne.dm]:
```suggestion
	for(var/obj/item/reagent_containers/syringe/dart as anything in underbarrel.syringes)
```

File: modular_skyrat/modules/medical/code/smartdarts.dm (hunk 3 above)
Comment (softcerv) [reply modular_skyrat/modules/medical/code/smartdarts.dm]: what does `as anything` do?

Comment (tf-4) [reply modular_skyrat/modules/medical/code/smartdarts.dm]: Skips the typecheck when iterating through the list of syringes. As there won't be anything _other_ than `.../syringe/dart` in there, it's unnecessary to check the type of each one.
So it's a performance gain. Very minor, but this is used all over the codebase and they add up.

Comment (softcerv) [reply modular_skyrat/modules/medical/code/smartdarts.dm]: gotcha

**Pull request merged**